e2e/solve/test.go

```diff
@@ -384,71 +401,81 @@ func openOrder(ctx context.Context, backends ethbackend.Backends, order TestOrde
+		}
 
-			req, err := http.NewRequestWithContext(ctx, http.MethodPost, url, bytes.NewBuffer(body))
-			if err != nil {
-				return errors.Wrap(err, "new request")
-			}
+		body, err := json.Marshal(checkReq)
+		if err != nil {
+			return errors.Wrap(err, "marshal request")
+		}
 
-			resp, err := http.DefaultClient.Do(req)
-			if err != nil {
-				return errors.Wrap(err, "do request")
-			}
+		req, err := http.NewRequestWithContext(ctx, http.MethodPost, url, bytes.NewBuffer(body))
+		if err != nil {
+			return errors.Wrap(err, "new request")
+		}
 
-			defer resp.Body.Close()
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			return errors.Wrap(err, "do request")
+		}
 
-			if resp.StatusCode != http.StatusOK {
-				return errors.New("bad status", "status", resp.StatusCode)
-			}
+		if resp.StatusCode != http.StatusOK {
+			return errors.New("bad status", "status", resp.StatusCode)
+		}
 
-			var checkResp solver.CheckResponse
-			if err = json.NewDecoder(resp.Body).Decode(&checkResp); err != nil {
-				return errors.Wrap(err, "decode response")
-			}
+		var checkResp solver.CheckResponse
+		if err = json.NewDecoder(resp.Body).Decode(&checkResp); err != nil {
+			return errors.Wrap(err, "decode response")
+		}
 
-			if checkResp.Rejected != order.ShouldReject {
-				return errors.New("unexpected rejection",
-					"expected", order.ShouldReject,
-					"actual", checkResp.Rejected,
-					"reason", checkResp.RejectReason,
-					"description", checkResp.RejectDescription,
-					"idx", i,
-				)
-			}
+		if err = resp.Body.Close(); err != nil {
+			return errors.Wrap(err, "close response body")
+		}
 
-			if checkResp.RejectReason != order.RejectReason {
-				return errors.New("unexpected reject reason", "expected", order.RejectReason, "actual", checkResp.RejectReason)
-			}
+		if checkResp.Rejected != order.ShouldReject {
+			return errors.New("unexpected rejection",
+				"expected", order.ShouldReject,
+				"actual", checkResp.Rejected,
+				"reason", checkResp.RejectReason,
+				"description", checkResp.RejectDescription,
+				"order_idx", i,
+			)
+		}
 
-			if checkResp.Accepted && order.ShouldReject {
-				return errors.New("accepted but should reject")
-			}
+		if checkResp.RejectReason != order.RejectReason {
+			return errors.New("unexpected reject reason", "expected", order.RejectReason, "actual", checkResp.RejectReason)
+		}
 
-			return nil
-		})
-	}
+		if checkResp.Accepted && order.ShouldReject {
+			return errors.New("accepted but should reject")
+		}
 
-	if err := eg.Wait(); err != nil {
-		return errors.Wrap(err, "wait checks")
+		// Refund solver native balance after test logic.
+		if isInsufficientInventory(order) {
+			eth1m := math.NewInt(1_000_000).MulRaw(params.Ether).BigInt()
+			if err := setSolverAccountNativeBalance(ctx, backends, eth1m); err != nil {
+				return errors.Wrap(err, "refund solver account failed")
+			}
+			log.Debug(ctx, "Solver account native balance refunded successfully")
```

Comment:
I would remove log, it feels like debug logging only required for you adding the test.